[Merged by Bors] - CI for New System Tests

web3rover opened the pull request:
This PR is to run the new system tests in the CI. The new system tests are already merged to develop. It should run the tests during `bors try` and `bors merge` in our GCP kubernetes infra.

Tasks:

- [x]  It builds the go-spacemesh and system tests docker images and pushes to dockerhub
- [x]  It runs the tests in a k8s cluster hosted on GCP
- [x]  After running the tests it prints the tests results and cleans all pods
- [x]  Run system tests only when bors try or bors merge is invoked
- [x]  Fail the CI when tests fail (read the status of kubernetes pod)

web3rover committed:
test gcp auth

web3rover committed:
added os

web3rover committed:
added cluter list

web3rover committed:
added kubectl auth

web3rover committed:
added manifest for k8s setup

web3rover committed:
create node pool

web3rover committed:
added zone

web3rover committed:
change commit

web3rover committed:
WIP

web3rover committed:
create and delete node pool

web3rover committed:
added machine type

web3rover committed:
changed machine type

web3rover committed:
added yes

web3rover committed:
added tests

web3rover committed:
build and push docker

web3rover committed:
WIP

web3rover committed:
WIP

web3rover committed:
WIP

web3rover committed:
WIP

web3rover committed:
WIP

web3rover committed:
WIP

web3rover committed:
WIP

web3rover committed:
WIP

web3rover committed:
WIP

web3rover committed:
WIP

web3rover committed:
Merge branch 'develop' into new_tests

web3rover committed:
WIP

web3rover committed:
Merge branch 'new_tests' of github.com:spacemeshos/go-spacemesh into new_tests

web3rover committed:
WIP

web3rover committed:
WIP

web3rover committed:
WIP

web3rover committed:
added old tests

web3rover committed:
changed dir name

web3rover committed:
changed name

web3rover committed:
Merge branch 'develop' into new_tests

web3rover committed:
changed name

web3rover committed:
Merge branch 'new_tests' of github.com:spacemeshos/go-spacemesh into new_tests

dshulyak commented:
@narayanprusty i merged pull request with tests. note that the module is `systest` (without s on the end).

dshulyak commented on .github/workflows/systest-ci.yml:
```diff
@@ -0,0 +1,71 @@
+      run: cd systests && make push
+
+    - name: Run tests
+      run: cd systests && make run test_name=TestSmeshing
```
this is what i am using `make run test_name=. smesher_image=spacemeshos/go-spacemesh-dev:develop size=50 bootstrap=10m level=info clusters=2 node_selector=cloud.google.com/gke-nodepool=dmitry-ci-preemtible-1`

dshulyak replied:
`clusters` controls how many clusters of `size` will be deployed concurrently. pool that is specified with `node-selector` should have enough resources to schedule pods on them.

web3rover committed:
Merge branch 'develop' into new_tests

web3rover committed:
updated make run

web3rover committed:
Merge branch 'develop' into new_tests

web3rover committed:
changed node pool

web3rover committed:
Merge branch 'new_tests' of github.com:spacemeshos/go-spacemesh into new_tests

web3rover committed:
removed random

web3rover committed:
Merge branch 'develop' into new_tests

web3rover committed:
added artifacts

web3rover committed:
WIP

web3rover committed:
fixed tortoise

dshulyak commented on systest/artifacts/readme.md:
```diff
@@ -0,0 +1,23 @@
+
+helm install grafana grafana/grafana --values ./grafana.yaml
+helm install loki grafana/loki --values ./loki.yaml
+helm install promtail grafana/promtail --values ./promtail.yaml
```
i was using chart `grafana/loki-stack ` 

```
helm upgrade --install loki grafana/loki-stack  --set grafana.enabled=true,prometheus.enabled=true,prometheus.alertmanager.persistentVolume.enabled=false,prometheus.server.persistentVolume.enabled=false,loki.persistence.enabled=true,loki.persistence.storageClassName=standard,loki.persistence.size=20Gi -f promtail.yaml
```

it automatically adds loki source to the grafana

what about cri-o pipeline, did you enable it or logs collector works without it?

web3rover replied:
The collector seems to work without the cri-o pipeline.

dshulyak reviewed (commented):
one general note is that artifacts should not be added to this repository, lets create a separate repo (for example `ci-infra`) with automation (using terraform as you mentioned)

web3rover committed:
removed artifacts

web3rover commented:
@dshulyak I have removed artefacts from this repo. I have created a separate repo and added the artifacts there. I will create a terraform playbook which uses grafana/loki-stack in that repo.

dshulyak commented:
@narayanprusty i realized that we should run new ci job only on `bors try` and `bors merge`. otherwise, it should be skipped as the rest of the long-running jobs.

web3rover committed:
make bors only

web3rover committed:
Merge branch 'develop' into new_tests

web3rover commented:
bors try

web3rover committed:
WIP

web3rover committed:
Merge branch 'new_tests' of github.com:spacemeshos/go-spacemesh into new_tests

web3rover committed:
WIP

narayanprusty commented:
bors try

web3rover commented:
bors try

web3rover committed:
WIP

web3rover commented:
bors try

web3rover committed:
Merge branch 'develop' into new_tests

web3rover committed:
handle tests failure

web3rover commented:
bors try

web3rover committed:
WIP

web3rover committed:
WIP

web3rover committed:
WIP

web3rover committed:
WIP

web3rover committed:
WIP

web3rover commented:
bors try

web3rover committed:
WIP

web3rover committed:
WIP

web3rover committed:
WIP

web3rover committed:
WIP

web3rover committed:
WIP

web3rover committed:
WIP

web3rover committed:
WIP

web3rover committed:
WIP

web3rover committed:
WIP

web3rover committed:
WIP

web3rover committed:
WIP

web3rover commented:
bors try

web3rover committed:
WIP

web3rover commented:
bors try

web3rover commented:
bors try

web3rover commented:
bors try

dshulyak commented on .github/workflows/ci.yml:
```diff
@@ -158,6 +158,73 @@ jobs:
+
+    - name: Tests Passed or Failed
+      run: |
+        LOGS='${{ steps.tests_logs.outputs.content }}'
```
i think you should check pod state instead. if program exits with any non zero code pod state should be failed

dshulyak replied:
e.g. we need to fail when tests can't compile

web3rover committed:
WIP

web3rover commented:
bors try

web3rover committed:
WIP

web3rover commented:
bors try

web3rover commented:
bors try

web3rover commented:
bors merge

web3rover committed:
skip partition test

web3rover committed:
WIP

web3rover committed:
WIP

web3rover committed:
WIP

web3rover commented:
bors merge

web3rover commented:
bors try
